.github/workflows/label-check.yaml

```
@@ -4,8 +4,10 @@ on:
   pull_request:
     types:
       - opened
+      - reopened
       - labeled
       - unlabeled
+      - synchronize
```

Re, `synchronize`:

> A synchronize pull_request event it’s only triggered when the pull request’s tracking branch is synchronized with the source branch for the pull request, and that happens when the source branch is updated.
> 
> So every time someone pushes to the source branch of the pull request, the workflow will kick off again.

From https://github.com/github/docs/issues/2257#issuecomment-772884922

Do we really need to run this on every push? That seems wasteful to me.

---

```suggestion
```

---

Ah, 

> I think we want it to run for all so we can add it to the requirements for merge-when-green

from @larsoner in https://github.com/numpy/numpydoc/pull/493/files#r1298430414.

In that case, should we maybe do something like this?

```suggestion
      # Enable if label is a requirement and auto-merge is enabled
      # - synchronize  
```

---

I am not sure we need to run it on every push, but we have had it mysteriously not running from time to time. I hadn't paid attention, but I think it involved PRs with multiple commits.

There is no rush on merging this, so I am happy to hear from others. We should also start paying closer attention to when the existing events don't cover things.

---

If you don't add it to merge-when-green requirements it's really easy to miss -- I did on the numpydoc pull request. So I'd expect most repos nowadays to prefer the `synchronize` event.

---

Fine with me. Just wanted to check that we are not using resources that aren't really needed.

---

The `synchronize` event may be overkill: we do not need to check the label on each commit. But it's true what Jarrod said about some PRs mysteriously not running the label check, so I suppose it's better to be conservative.

---

> The synchronize event may be overkill: we do not need to check the label on each commit.

To me the use case most people will want for this is "automatically require labels, make the PR red if it doesn't have them, and make it green otherwise". If you don't have it run on every commit and I make a PR, then push *one additional commit*, that PR will be green even without any label (because the most recent commit won't trigger the run). Okay so then maybe you decide to mark the labeler as a "required check" in the branch protection rules -- now it shows up as "expected" and orange on my second commit, but never changes from orange -- even when I push additional commits -- until someone relabels the PR manually, triggering a run. (I know this flow because this is what I just lived through in sphinx-gallery when I did it without the `synchronize` event!)

So AFAIK the only real way to get a reliable "green means I can merge knowing this is labeled" is to have it run on the "synchronize" event, which I assume is the behavior most people want.

---

Ah, okay, so the status does not live beyond a single commit :/  In that case, we have no choice. Thanks for explaining!

---

I guess another question then is whether this still needs the labeled/unlabeled etc. states.

---

I think so, because if I open a PR that's perfect other than a label and see red, then ask a maintainer to label, and they do, it's nice to have the PR magically turn green for me without needing to push any commits.

And for safety, if someone removes a label, it should turn red again